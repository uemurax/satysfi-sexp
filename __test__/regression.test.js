const shell = require("shelljs");
const tmp = require("tmp");

shell.cd("__test__");

afterAll(() => {
    shell.rm("test.pdf", "test.satysfi-aux");
});

test("SATySFi is installed", () => {
    expect(shell.exec("satysfi --version").code).toBe(0);
});

test("Compiler outputs", () => {
    const compilerOutput = shell
          .exec("satysfi test.saty")
          .exec("awk '/evaluating texts .../{flag=1;next}/evaluation done/{flag=0}flag'")
          .stdout;
    expect(compilerOutput).toMatchSnapshot();
});
