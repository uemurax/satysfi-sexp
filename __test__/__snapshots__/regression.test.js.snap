// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiler outputs 1`] = `
"[INFO] ========SExp-StdAtom========
[INFO] s1 = abc
[INFO] s2 = some-SYMBOL_name-4-2
[INFO] s3 = symbol-name-1
[INFO] ch-l3 =  symbol-name-2
[INFO] s6 = 334
[INFO] s7 = -13209542
[INFO] ========SExp========
[INFO] s8 = symbol-1
[INFO] s9 = (symbol-1 (\\"strin\\\\\\"g\\\\\\"\\" -334))
[INFO] s10 = (this (is \\"an\\") (S-expression 309 -334 \\"\\\\\\"!\\\\\\\\!\\\\\\"\\"))
"
`;
